src/runner.rs

```diff
@@ -46,79 +94,145 @@ pub async fn run_test_command(
                     );
                     continue;
                 }
-                let file = match std::fs::File::open(&file.original_path) {
-                    Ok(file) => file,
-                    Err(e) => {
-                        log::warn!("Error opening file: {}", e);
-                        continue;
-                    }
-                };
-                let reader = std::io::BufReader::new(file);
-                let junitxml = match junit_parser::from_reader(reader) {
-                    Ok(junitxml) => junitxml,
-                    Err(e) => {
-                        log::warn!("Error parsing junitxml: {}", e);
-                        continue;
-                    }
-                };
-                for suite in junitxml.suites {
-                    let parent_name = suite.name;
-                    for case in suite.cases {
-                        let failure = case.status.is_failure();
-                        if failure {
-                            let name = case.original_name;
-                            log::debug!("Test failed: {} -> {}", parent_name, name);
-                            failures.push(Test {
-                                parent_name: parent_name.clone(),
-                                name: name.clone(),
-                                class_name: case.classname.clone(),
-                                file: case.file.clone(),
-                            });
-                        }
+            }
+            let file = match std::fs::File::open(&file.original_path) {
+                Ok(file) => file,
+                Err(e) => {
+                    log::warn!("Error opening file: {}", e);
+                    continue;
+                }
+            };
+            let reader = std::io::BufReader::new(file);
+            let junitxml = match junit_parser::from_reader(reader) {
+                Ok(junitxml) => junitxml,
+                Err(e) => {
+                    log::warn!("Error parsing junitxml: {}", e);
+                    continue;
+                }
+            };
+            for suite in junitxml.suites {
+                let parent_name = suite.name;
+                for case in suite.cases {
+                    let failure = case.status.is_failure();
+                    if failure {
+                        let name = case.original_name;
+                        log::debug!("Test failed: {} -> {}", parent_name, name);
+                        failures.push(Test {
+                            parent_name: parent_name.clone(),
+                            name: name.clone(),
+                            class_name: case.classname.clone(),
+                            file: case.file.clone(),
+                        });
                     }
                 }
             }
         }
     }
-    let exit_code = result.code().unwrap_or(EXIT_FAILURE);
-    Ok(RunResult {
+    Ok(failures)
+}
+
+pub async fn run_quarantine(
```

Comment:
It looks a bit confusing that `run_quarantine` accepts `quarantining` flag. To me it sounds that `quarantining` should control if `run_quarantine` is called.

Reply:
I wanted to avoid duplicating around this logic, and the default values

Reply:
I would still move it outside. There is only one place where we check `use_quarantining` and it's more clear for the reader to have it in main function.